metaflow/plugins/kfp/s3_sensor_decorator.py

```diff
@@ -0,0 +1,109 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+
+from types import FunctionType
+from typing import Tuple
+
+"""
+Within identity_formatter, which is passed in as the func parameter,
+customers have access to all variables in flow_parameters_json (which 
+include all parameters passed in through the flow at compile and run time by
+KFP) as well as a number of environment variables, including:
+
+MF_POD_NAME
+MF_POD_NAMESPACE
+MF_ARGO_NODE_NAME
+MF_ARGO_WORKFLOW_NAME
+ZODIAC_SERVICE
+ZODIAC_TEAM
+POD_NAMESPACE
+KFP_SDK_NAMESPACE
+POD_NAME
+POD_IP
+ARGO_WORKFLOW_NAME
+K8S_CLUSTER_NAME
+K8S_CLUSTER_ENV
+METAFLOW_NOTIFY_EMAIL_SMTP_HOST
+METAFLOW_NOTIFY_EMAIL_SMTP_PORT
+METAFLOW_NOTIFY_EMAIL_FROM
+METAFLOW_NOTIFY_EMAIL_BODY
+METAFLOW_NOTIFY_ON_ERROR
+KFP_RUN_URL_PREFIX
+METAFLOW_DEFAULT_DATASTORE
+METAFLOW_DEFAULT_METADATA
+METAFLOW_SERVICE_URL
+METAFLOW_DATASTORE_SYSROOT_S3
+WAGGLE_DANCE_DEFAULT_ENVIRONMENT
+WAGGLE_DANCE_DEV_ENDPOINT
+WAGGLE_DANCE_STAGE_ENDPOINT
+WAGGLE_DANCE_PROD_READ_ONLY_ENDPOINT
+WAGGLE_DANCE_PROD_ENDPOINT
+AWS_ROLE_ARN
+AWS_WEB_IDENTITY_TOKEN_FILE
+"""
+
+
+def identity_formatter(path: str, flow_parameters_json: dict) -> str:
+    return path
+
+
+"""
+@s3_sensor is implemented as a flow decorator that's used by customers to ensure 
+a workflow only begins after a key in S3 has been written to. 
+
+Example usage:
+
+    def formatter(path: str, flow_parameters_json: dict) -> str:
+        return path.replace("FLOW_ID", flow_parameters_json["FLOW_ID"])
+
+    @s3_sensor(
+        path="s3://aip-example-sandbox/metaflow/S3SensorFlow/data/$date/61/FLOW_ID",
+        timeout_seconds=3600, # 1 hour
+        polling_interval_seconds=90,
+        formatter=formatter
+    )
```

Comment:
Can you provide an example without a "formatter" while accessing `flow_parameters_json` ?

Reply:
Especially accessing `scheduled_time`

Reply:
Why would the user want to access `flow_parameters_json` outside of a "formatter" func? What is the use case?

I thought the whole point of "formatter" func was to access `flow_parameters_json` at runtime.

Reply:
The point of formatter function is to put in fields that are computed at runtime.  Else you don't need a function, if it can all be declarative.